File: core/src/com/colorflow/ring/SideTapListener.java
package com.colorflow.ring;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.InputAdapter;
import com.colorflow.utility.Position;

class SideTapListener extends InputAdapter implements RingListener {

    private Ring ring;
    private float rot;

    SideTapListener(Ring ring) {
        this.ring = ring;
    }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        calcRot();
        return false;
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        calcRot();
        return false;
    }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
        calcRot();
        return false;
    }

    @Override
    public void onRingAct() {
        ring.rotateBy(rot);
    }

    private void calcRot() {
        int clock = 0;
        for (int i = 0; i < 10; i++) {
            if (Gdx.input.isTouched(i)) {
                if (Gdx.input.getX(i) > Position.getCenter().getX()) {
                    clock++;
                } else {
                    clock --;
                }
            }
        }
        rot = clock * ring.getSensibility();
    }
}

File: core/src/com/colorflow/ring/SideTapListener.kt
package com.colorflow.ring

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.InputAdapter
import com.colorflow.utility.Position

internal class SideTapListener(private val ring: Ring) : InputAdapter(), RingListener {
    private var rot: Float = 0.toFloat()

    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean {
        calcRot()
        return false
    }

    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean {
        calcRot()
        return false
    }

    override fun touchUp(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean {
        calcRot()
        return false
    }

    override fun onRingAct() {
        ring.rotateBy(rot)
    }

    private fun calcRot() {
        var clock = 0
        for (i in 0..9) {
            if (Gdx.input.isTouched(i)) {
                if (Gdx.input.getX(i) > Position.center.x) {
                    clock++
                } else {
                    clock--
                }
            }
        }
        rot = clock * ring.sensibility
    }
}